### PR TITLE
Add .travis.yml

### PR DESCRIPTION
**Pull request by ryanewel**
*Issue #, if available:*

*Description of changes:*
Updating .travis.yml to test build with colcon

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+  - sudo apt-get update -qq
+  - sudo apt-get install dpkg
+  - sh -c "echo $ROS_CI_DESKTOP"
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-ros-base
+  - source /opt/ros/$ROS_DISTRO/setup.bash
+  # Prepare rosdep to install dependencies.
+  - sudo rosdep init
 install:
-  - git clone https://github.com/ros-industrial/industrial_ci.git robot_ws/.ros_ci
+  - sudo ./scripts/setup.sh
```
Comment (mm318) [inline .travis.yml]: Can you use "make setup" instead?

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
 script:
-  - cd robot_ws && .ros_ci/travis.sh
+  - sudo sh -c "./scripts/build.sh robot_ws"
+  - sudo sh -c "./scripts/build.sh simulation_ws"
```
Comment (mm318) [inline .travis.yml]: Can you use "make build" instead of these two lines?

File: .travis.yml [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
-    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu ROS_CI_DESKTOP="`lsb_release -cs`"
+before_install:
+  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
+  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+  - sudo apt-get update -qq
+  - sudo apt-get install dpkg
+  - sh -c "echo $ROS_CI_DESKTOP"
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-ros-base
```
Comment (mm318) [inline .travis.yml:20]: Why not use industrial_ci.git anymore? Doesn't that handle installing ROS and rosdep for you? @AAlon

Comment (ryanewel) [reply .travis.yml:20]: My reasoning was that industrial_ci uses catkin make while it's recommended that the sample application is built with colcon

Comment (mm318) [reply .travis.yml:20]: Alright, sounds good.

File: .travis.yml [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
-  - gcc
+language: generic
+dist: xenial
 notifications:
   email:
     on_success: change
     on_failure: always
     recipients:
       - ros-contributions@amazon.com
+      - travis-build@platform-notifications.robomaker.aws.a2z.com
```
Comment (mm318) [inline .travis.yml:10]: According to @juanrh this line caused this travis build failure https://travis-ci.org/aws-robotics/kinesisvideo-common/jobs/476253478. Let's wait to see what the resolution should be.

File: .travis.yml [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-ros-base
+  - source /opt/ros/$ROS_DISTRO/setup.bash
+  # Prepare rosdep to install dependencies.
+  - sudo rosdep init
 install:
-  - git clone https://github.com/ros-industrial/industrial_ci.git robot_ws/.ros_ci
+  - sudo make setup
 script:
-  - cd robot_ws && .ros_ci/travis.sh
+  - sudo make build
```
Comment (mm318) [inline .travis.yml]: Have you tried without `sudo`?

Comment (ryanewel) [reply .travis.yml]: Removed sudo

File: .travis.yml (hunk 4 above)
Comment (mm318) [reply .travis.yml:10]: Okay, looks like it's not a issue with `.travis.yml`. Thanks.

File: .travis.yml (hunk 5 above)
Comment (mm318) [reply .travis.yml]: Great!

**Review by mm318 — APPROVED** [review]

**Pull request merged**